test/jdk/javax/swing/plaf/basic/BasicTextUI/8001470/bug8001470.java

```diff
@@ -26,45 +26,76 @@
  * @key headful
  * @bug 8001470
  * @summary JTextField's size is computed incorrectly when it contains Indic or Thai characters
- * @author Semyon Sadetsky
  */
 
-import javax.swing.*;
-import java.awt.*;
+import java.awt.Font;
+import java.awt.GraphicsEnvironment;
+import java.awt.GridLayout;
+import javax.swing.JFrame;
+import javax.swing.JTextField;
+import javax.swing.SwingUtilities;
+import javax.swing.JPanel;
 
 public class bug8001470 {
 
     private static JFrame frame;
     private static JTextField textField1;
     private static JTextField textField2;
+    private static boolean fontFound = false;
+
+    private static volatile int height1;
+    private static volatile int height2;
 
     public static void main(String[] args) throws Exception {
-        SwingUtilities.invokeAndWait(new Runnable() {
-            @Override
-            public void run() {
-                frame = new JFrame("JTextField Test");
-                frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        try {
+            SwingUtilities.invokeAndWait(new Runnable() {
+                @Override
+                public void run() {
+                    GraphicsEnvironment ge = GraphicsEnvironment.getLocalGraphicsEnvironment();
+                    String[] names = ge.getAvailableFontFamilyNames();
+                    Font[] allFonts = ge.getAllFonts();
+                    for (int x = 0; x < allFonts.length; x++) {
+                        if ((allFonts[x].canDisplay('\u0e01')) && (allFonts[x].canDisplay('\u0c01'))) {
```

Comment:
You may want to declare the characters as constants so that you create text fields with the characters you're searching the font for.